Documentation/WorkflowSwiftUI Adoption Guide.md

```diff
@@ -0,0 +1,448 @@
+
+```swift
+@ObservableState
+public struct State {
+  var foo: String
+  var bar: Int
+}
+```
+
+For an `ObservableState` type, creating a copy does not change the state’s identity in the copy. However, creating a new value by calling the initializer will result in a new identity.
+
+If your state has nested types, you can annotate them with `@ObservableState` as well, for fine-grained observation of the properties of that type as well.
+
+```swift
+@ObservableState
+public struct State {
+  var detail: Detail
+}
+
+@ObservableState
+public struct Detail {
+  var name: String
+  var age: Int
+}
+```
+
+In this example, changes to `state.detail.name` do not affect views that only use `state.detail.age` , and vice-versa. Without annotating `Detail` as `@ObservableState`, any change to a property on `state.detail` will invalidate views using any other property.
+
+## Rendering models
+
+Your workflow should render a type conforming to `ObservableModel`. This provides your view with a way to read from state and send updates that follow Workflow’s unidirectional data flow.
+
+```swift
+public protocol ObservableModel<State> {
+    /// The associated state type that this model observes.
+    associatedtype State: ObservableState
+
+    /// The accessor that can be used to read and write state.
+    var accessor: StateAccessor<State> { get }
+}
+```
+
+Some conveniences are available on the `RenderContext` to make it easy to create a model type.
+
+For trivial workflows with no actions, you can generate a model directly from your state:
+
+```swift
+struct TrivialWorkflow: Workflow {
+    typealias Output = Never
+
+    @ObservableState
+    struct State {
+        var counter = 0
+    }
+
+    func makeInitialState() -> State {
+        .init()
+    }
+
+    func render(
+        state: State,
+        context: RenderContext<Self>
+    ) -> StateAccessor<State> {
+        context.makeStateAccessor(state: state)
+    }
+}
+
+```
+
+For simple workflows with a single action, you can generate a model from your state and action:
+
+```swift
+struct SingleActionWorkflow: Workflow {
+    typealias Output = Never
+
+    @ObservableState
+    struct State {
+        var counter = 0
+    }
+
+    enum Action: WorkflowAction {
+        typealias WorkflowType = SingleActionWorkflow
+        case increment
+
+        func apply(toState state: inout State) -> Never? {
+            state.counter += 1
+            return nil
+        }
+    }
+
+    func makeInitialState() -> State {
+        .init()
+    }
+
+    func render(
+        state: State,
+        context: RenderContext<Self>
+    ) -> ActionModel<State, Action> {
+        context.makeActionModel(state: state)
+    }
+}
+
+```
+
+For complex workflows that have multiple actions or compose observable models from child
```

Comment:
> For complex workflows that have multiple actions

I feel like this is basically _the_ common case, it'd be nice to make this require less boilerplate/code to be written.